Realm.Shared/Dynamic/MetaRealmObject.cs

```diff
@@ -50,64 +51,63 @@ public override DynamicMetaObject BindGetMember(GetMemberBinder binder)
             var arguments = new List<Expression>
             {
                 WeakConstant(_metadata.Table),
-                Expression.Field(GetLimitedSelf(), RealmObjectRowHandleField),
-                Expression.Constant(_metadata.ColumnIndices[property.Name])
+                Expression.Constant(_metadata.ColumnIndices[property.Name]),
```

Comment:
Could we start documenting **why** we do some of these things with comments?
This is complex stuff for somebody to come back to or new to the team to learn.